### PR TITLE
connectionTimeout option on ChatManager constructor

### PR DESCRIPTION
**Pull request by callum-oakley**
Make the connection timeout configurable with the `connectionTimeout` option to the `ChatManager` constructor.

see discussion in #80

### CODE REVIEW

**Comment by callum-oakley** [comment]
corresponding mimir change: https://github.com/pusher/mimir/pull/406

**Comment by callum-oakley** [comment]
corresponding spec change: https://github.com/pusher/chatkit-spec/pull/15

File: src/constants.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
 export const TYPING_INDICATOR_TTL = 1500
 export const TYPING_INDICATOR_LEEWAY = 500
 export const SET_CURSOR_WAIT = 500
-export const SUBSCRIPTION_TIMEOUT = 5000
+export const CONNECTION_TIMEOUT = 10 * 1000
```
Comment (vivangkumar) [inline src/constants.js]: Perhaps we should call this `DEFAULT_CONNECTION_TIMEOUT`?

Comment (callum-oakley) [reply src/constants.js]: good idea. Makes it clear that it can be changed, unlike the others.

**Pull request merged**